### ISSUE TITLE
[Bug][Config UI] Blueprint's setting page throws error

### ISSUE THREAD

**Issue by hezyin** [apache/incubator-devlake#3313]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

When I visit the "settings" tab from the blueprint page,  config UI throws an application error:

<img width="2261" alt="Screen Shot 2022-10-07 at 5 37 35 PM" src="https://user-images.githubusercontent.com/2908155/194678774-a0c3d161-c25d-47c3-a71b-d0715ddeb154.png">

<img width="750" alt="Screen Shot 2022-10-07 at 5 37 20 PM" src="https://user-images.githubusercontent.com/2908155/194678752-09c2bbaf-ca64-4d34-91f4-7fdedf0ed3c2.png">

### What you expected to happen

Config ui renders the settings page successfully.

### How to reproduce

**Version: v0.14.0-alpha1**
Installation method: docker-compose

The blueprint was created using API:

```
curl --location --request POST 'http://localhost:8080/blueprints' \
--header 'Content-Type: application/json' \
--data-raw '{
    "enabled": true,
    "mode": "NORMAL",
    "name": "My Bitbucket Blueprint",
    "cronConfig": "",
    "isManual": true,
    "plan": [[]],
    "settings": {
        "connections": [
            {
                "plugin": "bitbucket",
                "connectionId": 2,
                "scope": [
                    {
                        "entities": [
                            "CODE",
                            "TICKET",
                            "CODEREVIEW",
                            "CROSS"
                        ],
                        "options": {
                            "owner": "yinhezheng",
                            "repo": "incubator-devlake"
                        }
                    }
                ]
            }
        ],
        "version": "1.0.0"
    }
}'
```

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by mintsweet** [comment]
I'll fix it.

**Comment by mintsweet** [comment]
I tested it and did not reproduce this issue. It may be because this blueprint was created before. This issue was caused by adjusting the displayed name in the settings. You can try to recreate a blueprint.

**Comment by hezyin** [comment]
@mintsweet I tried v0.14.0 image and confirmed I can reproduce the issue with the following steps:

1. Create a Bitbucket data connection with

```
curl --location --request POST 'http://localhost:8080/plugins/bitbucket/connections' \
--header 'Content-Type: application/json' \
--data-raw '{
    "endpoint": "https://api.bitbucket.org/2.0/",
    "username": "yinhezheng",
    "password": "xxxxxxxxxxxxx",
    "name": "Bitbucket Cloud"
}'
```

2. Create a new blueprint for Bitbucket

```
curl --location --request POST 'http://localhost:8080/blueprints' \
--header 'Content-Type: application/json' \
--data-raw '{
    "enabled": true,
    "mode": "NORMAL",
    "name": "My Bitbucket Blueprint",
    "cronConfig": "",
    "isManual": true,
    "plan": [[]],
    "settings": {
        "connections": [
            {
                "plugin": "bitbucket",
                "connectionId": 1,
                "scope": [
                    {
                        "entities": [
                            "CODE",
                            "TICKET",
                            "CODEREVIEW",
                            "CROSS"
                        ],
                        "options": {
                            "owner": "yinhezheng",
                            "repo": "incubator-devlake"
                        }
                    }
                ]
            }
        ],
        "version": "1.0.0"
    }
}'
```

3. Then when you go to "settings" of the newly created blueprint, you'll encounter the application error.

**Comment by klesh** [comment]
#3341 is a valid fix, however, it reveal the nature of the problem: the API lacks the ability to detect invalid request and failed to validate the user input.
@likyh would look into it and fix the BE problem.

**Comment by hezyin** [comment]
@klesh Is the API request invalid because it doesn't have the `transformations` parameter? Can we make the `transformations` optional? It's less of a hassle for users who just want to collect data and don't want to worry about data transformations at least at the begining.

P.S. Did you mean #3343 is a valid fix instead of #3341?

**Comment by likyh** [comment]
It's out-of-time because of refactor.
